### PR TITLE
Delegated selection event handling responsibilities out of App.cs.  

### PR DESCRIPTION
**Pull request by comintern**
Ref #1258

### CODE REVIEW

File: RetailCoder.VBE/App.cs [hunk 1]
```diff
@@ -44,171 +34,27 @@ public sealed class App : IDisposable
...
             _configService.SettingsChanged += _configService_SettingsChanged;
-            _parser.State.StateChanged += Parser_StateChanged;
-            _parser.State.StatusMessageUpdate += State_StatusMessageUpdate;
-
+            
             UiDispatcher.Initialize();
         }
 
-        //TODO - This should be able to move to the appropriate handling classes now.
-        #region Statusbar
```
Comment (retailcoder) [inline RetailCoder.VBE/App.cs]: So long, farewell, auf wiederzehn!

File: RetailCoder.VBE/App.cs [hunk 2]
```diff
@@ -44,171 +34,27 @@ public sealed class App : IDisposable
 
         public App(IVBE vbe, 
             IMessageBox messageBox,
-            IParseCoordinator parser,
             IGeneralConfigService configService,
             IAppMenu appMenus,
             RubberduckCommandBar stateBar,
             IRubberduckHooks hooks,
             IVersionCheck version,
```
Comment (retailcoder) [inline RetailCoder.VBE/App.cs]: I need to turn the version check into a *menu* command, and remove the `IVersionCheck` and `CommandBase` parameters - then we're down to 5 and that would be my upper limit for the number of dependencies a cohesive class might need to have.

File: RetailCoder.VBE/AppMenu.cs [hunk 3]
```diff
@@ -10,10 +12,17 @@ namespace Rubberduck
     public class AppMenu : IAppMenu, IDisposable
     {
         private readonly IReadOnlyList<IParentMenuItem> _menus;
+        private readonly IParseCoordinator _parser;
+        private readonly ISelectionChangeService _selectionService;
 
-        public AppMenu(IEnumerable<IParentMenuItem> menus)
+        public AppMenu(IEnumerable<IParentMenuItem> menus, IParseCoordinator parser, ISelectionChangeService selectionService)
```
Comment (retailcoder) [inline RetailCoder.VBE/AppMenu.cs]: 👍

File: RetailCoder.VBE/App.cs (hunk 2 above)
Comment (comintern) [inline RetailCoder.VBE/App.cs]: @retailcoder  - `configService` can go pretty soon too. It's really only needed for localization and hotkeys - App is holding the dependency as a "proxy" for classes that actually need to respond to its events.  There's no reason that App should be calling methods directly on classes that can subscribe to an event handler.

File: RetailCoder.VBE/App.cs (hunk 1 above)
Comment (Vogel612) [inline RetailCoder.VBE/App.cs]: s/z/s ... make that "auf nimmerwiedersehen" :+1: :wave:

**Pull request merged**